RoaringBitmap/src/main/java/org/roaringbitmap/Container.java

```diff
@@ -190,9 +190,23 @@ public Container andNot(Container x) {
    */
   public Container orNot(Container x, int endOfRange) {
     if (endOfRange < 0x10000) {
-      return or(x.not(0, endOfRange).iremove(endOfRange, 0x10000));
+      Container not = x.not(0, endOfRange);
+      Container removed = not.iremove(endOfRange, 0x10000);
+      if (not != removed) {
+        not.close();
+      }
+      Container answer = or(removed);
+      if (answer != removed) {
+        removed.close();
+      }
+      return answer;
+    }
+    Container not = x.not(0, 0x10000);
+    Container answer = or(not);
+    if (not != answer) {
+      not.close();
     }
-    return or(x.not(0, 0x10000));
+    return answer;
```

Comment:
This diff sums up the complexity tradeoff for me, it may even mess with inlining by increasing the code size (fortunately it's not a widely used method)